aws/request_pagination.go

```diff
@@ -141,13 +141,29 @@ func (r *Request) nextPageTokens() []interface{} {
 	tokens := []interface{}{}
 	tokenAdded := false
 	for _, outToken := range r.Operation.OutputTokens {
-		v, _ := awsutil.ValuesAtPath(r.Data, outToken)
-		if len(v) > 0 {
-			tokens = append(tokens, v[0])
-			tokenAdded = true
-		} else {
+		vs, _ := awsutil.ValuesAtPath(r.Data, outToken)
+
+		if len(vs) == 0 {
 			tokens = append(tokens, nil)
+			continue
+		}
+		v := vs[0]
+
+		switch tv := v.(type) {
+		case *string:
+			if len(StringValue(tv)) == 0 {
+				tokens = append(tokens, nil)
+				continue
+			}
+		case string:
+			if len(tv) == 0 {
+				tokens = append(tokens, nil)
+				continue
+			}
 		}
+
+		tokenAdded = true
```

Comment:
if they are bunch of empy strings, wouldn't `tokenAdded` not be set to `true`? Meaning this would never return the `tokens`

Reply:
`tokenAdded` is present if a valid token is added. The `tokens` slice is a sparse array. Its index directly correspond to indexes in another slice.

If all of the values at the `outTokens` paths are empty strings there will be no valid tokens in the `tokens` slice. In that case returning `nil` would be the correct action.  If there is at least one valid value in `outTokens` paths the `tokenAdded` will be set to true.